cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -746,9 +750,38 @@ func create(ctx context.Context, flags *pflag.FlagSet) error {
 		extraKernelArgs = procfs.NewCmdline(extraBootKernelArgs)
 	}
 
+	if withSiderolinkAgent {
+		if extraKernelArgs == nil {
+			extraKernelArgs = procfs.NewCmdline("")
+		}
+
+		if extraKernelArgs.Get("siderolink.api") != nil || extraKernelArgs.Get("talos.events.sink") != nil || extraKernelArgs.Get("talos.logging.kernel") != nil {
+			return errors.New("siderolink kernel arguments are already set, cannot run with --with-siderolink")
+		}
+
+		wgPort := rand.IntN(20000) + 40000
+		wgHost := gatewayIPs[0].String()
+		apiPort := strconv.Itoa(wgPort + 1)
+		sinkPort := strconv.Itoa(wgPort + 2)
+		logPort := strconv.Itoa(wgPort + 3)
+
+		request.SiderolinkRequest.WireguardEndpoint = net.JoinHostPort(wgHost, strconv.Itoa(wgPort))
+		request.SiderolinkRequest.APIEndpoint = ":" + apiPort
+		request.SiderolinkRequest.SinkEndpoint = ":" + sinkPort
+		request.SiderolinkRequest.LogEndpoint = ":" + logPort
+
+		apiLink := "grpc://" + net.JoinHostPort(wgHost, apiPort) + "?jointoken=foo"
+
+		extraKernelArgs.Append("siderolink.api", apiLink)
+		extraKernelArgs.Append("talos.events.sink", "[fdae:41e4:649b:9303::1]:"+sinkPort)
```

Comment:
probably makes sense to define this IP as a constant

Reply:
Done - we now use `GetAgentNodeAddr()` which method which uses generated prefix from `wireguard.NetworkPrefix("")`.